app/services/report_importer.rb

```diff
@@ -91,21 +91,60 @@ def logs
   end
 
   def import_log_messages
+    id_messages = []
```

Comment:
I think using hashes here with a reasonable key (i.e. message or source value) instead of arrays can both make the lookups faster and the code easier to understand.

Reply:
I don't actually search in arrays, I just append once and iterate once over them. If you have a better idea let me know how to do this.